plugins/pulsar-records-storage/src/test/java/com/github/bsideup/liiklus/pulsar/PulsarRecordsStorageTest.java

```diff
@@ -70,4 +91,212 @@ public String keyByPartition(int partition) {
     public int getNumberOfPartitions() {
         return NUM_OF_PARTITIONS;
     }
+
+    @Test
+    @Override
+    // since pulsar behave differently for closed ledger, need to overwrite this test to include the 1 entry setback
```

Comment:
since the whole point of Liiklus is too abstract away the semantics of different event storages, I don't understand why this test does not pass as it is in the TCK?

Also, TCK tests should not be changed, only new tests can be added. There are assumptions for storages that do not support more than 1 partition, but other than that, the storages should behave similarly

Reply:
For this one, it's indeed because there is an inconsistency in pulsar behaviour for checking what is a valid entry id. This would mean that we will need to patch pulsar and wait for a new release.

Reply:
FYI I'm trying to get help in Pulsar's Slack about it, will report back as soon as I have any info

Reply:
you can check the PulsarRecordsStorage comment on the adaptForSeek method.

Reply:
See https://apache-pulsar.slack.com/archives/C5Z4T36F7/p1567085799031200

Reply:
have no access.

Reply:
I would highly suggest [to get it](https://apache-pulsar.herokuapp.com/) if you plan to work with Pulsar, they have a very active community and help with different kinds of questions :)

Reply:
got it